src/experiments.rs

```diff
@@ -322,11 +360,70 @@ impl Experiment {
         .collect::<Fallible<Vec<Crate>>>()
     }
 
-    pub fn get_uncompleted_crates(&self, db: &Database) -> Fallible<Vec<Crate>> {
+    fn crate_list_size(&self, config: &Config) -> i32 {
+        match self.assigned_to {
+            //if experiment is distributed return chunk
+            Some(Assignee::Distributed) => config.chunk_size(),
+            //if experiment is assigned to specific agent return all the crates
+            _ => FULL_LIST,
+        }
+    }
+
+    pub fn get_uncompleted_crates(
+        &self,
+        db: &Database,
+        config: &Config,
+        assigned_to: &Assignee,
+    ) -> Fallible<Vec<Crate>> {
+        let limit = self.crate_list_size(config);
+
+        db.transaction(|transaction| {
+            //get the first 'limit' queued crates from the experiment crates list
+            let crates = transaction
+                .query(
+                    "SELECT crate FROM experiment_crates WHERE experiment = ?1
+                     AND status = ?2 AND skipped = 0 LIMIT ?3;",
+                    &[&self.name, &Status::Queued.to_string(), &limit],
+                    |r| {
+                        let value: String = r.get("crate");
+                        Ok(serde_json::from_str(&value)?)
+                    },
+                )?
+                .into_iter()
+                .collect::<Fallible<Vec<Crate>>>();
+
+            //update the status of the previously selected crates to 'Running'
+            transaction.execute(
+                "UPDATE experiment_crates SET assigned_to = ?1, status = ?2 \
+                 WHERE experiment = ?3 AND crate IN (SELECT crate FROM \
```

Comment:
Could we instead do `AND crate IN <list>` passing a list of the IDs previously selected?